test/gtest/core/test_mc_rocm.cc

```diff
@@ -0,0 +1,201 @@
+ * See file LICENSE for terms.
+ */
+
+extern "C" {
+#include <components/mc/ucc_mc.h>
+#include <pthread.h>
+}
+#include <common/test.h>
+#include <hip/hip_runtime.h>
+#include <vector>
+
+void *mt_ucc_mc_rocm_allocs(void *args)
+{
+    // Final size will be:
+    // size * (quantifier^(num_of_allocs/2))
+    // and should be larger than mpool buffer size which is 1MB by default,
+    // to assure testing both fast and slow ucc_mc_alloc path.
+    // if num_of_allocs is changed, change quantifier accordingly.
+    int                                   quantifier    = 2;
+    size_t                                size          = 4;
+    int                                   num_of_allocs = 40;
+    std::vector<ucc_mc_buffer_header_t *> headers;
+    std::vector<void *>                   pointers;
+    headers.resize(num_of_allocs);
+    pointers.resize(num_of_allocs);
+
+    for (int i = 0; i < num_of_allocs; i++) {
+        EXPECT_EQ(UCC_OK,
+                  ucc_mc_alloc(&headers[i], size, UCC_MEMORY_TYPE_ROCM));
+        pointers[i] = headers[i]->addr;
+        EXPECT_EQ(hipSuccess, hipMemset(pointers[i], 0, size));
+        EXPECT_EQ(UCC_OK, ucc_mc_free(headers[i]));
+        if (i % 2) {
+            size *= quantifier;
+        }
+    }
+    return 0;
+}
+
+class test_mc_rocm : public ucc::test {
+  protected:
+    const int               TEST_ALLOC_SIZE = 1024;
+    ucc_mc_buffer_header_t *mc_header;
+    void *                  test_ptr;
+    ucc_memory_type_t       test_mtype;
+    void TestMCRocmSetUp(ucc_mc_params_t mc_params)
+    {
+        ucc::test::SetUp();
+        ucc_constructor();
+        ucc_mc_init(&mc_params);
+        test_ptr   = NULL;
+        test_mtype = UCC_MEMORY_TYPE_UNKNOWN;
+    }
+    virtual void SetUp() override
+    {
+        ucc_mc_params_t mc_params = {
+            .thread_mode = UCC_THREAD_SINGLE,
+        };
+        TestMCRocmSetUp(mc_params);
+    }
+    virtual void TearDown() override
+    {
+        ucc_mc_finalize();
+        ucc::test::TearDown();
+    }
+};
+
+class test_mc_rocm_mt : public test_mc_rocm {
+  protected:
+    virtual void SetUp() override
+    {
+        ucc_mc_params_t mc_params = {
+            .thread_mode = UCC_THREAD_MULTIPLE,
+        };
+        TestMCRocmSetUp(mc_params);
+    }
+};
+
+UCC_TEST_F(test_mc_rocm, mc_rocm_load)
+{
+    EXPECT_EQ(UCC_OK, ucc_mc_available(UCC_MEMORY_TYPE_ROCM));
+}
+
+UCC_TEST_F(test_mc_rocm, can_alloc_and_free_mem)
+{
+    // mpool will be used only if size is smaller than UCC_MC_CPU_ELEM_SIZE, which by default set to 1MB and is configurable at runtime.
+    EXPECT_EQ(UCC_OK,
+              ucc_mc_alloc(&mc_header, TEST_ALLOC_SIZE, UCC_MEMORY_TYPE_ROCM));
+    test_ptr = mc_header->addr;
+    EXPECT_EQ(hipSuccess, hipMemset(test_ptr, 0, TEST_ALLOC_SIZE));
+    EXPECT_EQ(UCC_OK, ucc_mc_free(mc_header));
+}
+
+// Disabled because can't reinit mc with different thread mode
+UCC_TEST_F(test_mc_rocm_mt, DISABLED_can_alloc_and_free_mem_mt)
```

Comment:
@Sergei-Lebedev i think we can re-enable MT tests for mc, right? since we alway init MC in THREAD_MULTIPLE mode